test/sequential/market.nonTransferable.multiswap.test.ts

```diff
@@ -0,0 +1,294 @@
+    ),
+  ).then((events) => {
+    const res = getEventResultFromMangataTx(events);
+    expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+  foundationMembers = await FoundationMembers.getFoundationMembers();
+  expect(foundationMembers).toContain(testUser);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudo(api.tx.tokens.mint(token1, testUser, BN_HUNDRED_BILLIONS)),
+    Sudo.sudo(api.tx.tokens.mint(token2, testUser, BN_HUNDRED_BILLIONS)),
+    Sudo.sudo(api.tx.tokens.mint(token3, testUser, BN_HUNDRED_BILLIONS)),
+    Sudo.sudo(api.tx.tokens.mint(token4, testUser, BN_HUNDRED_BILLIONS)),
+    Sudo.sudo(api.tx.tokens.mint(token5, testUser, BN_HUNDRED_BILLIONS)),
+    Sudo.sudo(api.tx.tokens.mint(token6, testUser, BN_HUNDRED_BILLIONS)),
+  );
+  const poolEvents = await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(token1, BN_MILLION, token2, BN_MILLION, "StableSwap"),
+    ),
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(
+        token2,
+        BN_MILLION,
+        GASP_ASSET_ID,
+        BN_MILLION,
+        "StableSwap",
+      ),
+    ),
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(
+        GASP_ASSET_ID,
+        BN_MILLION,
+        token3,
+        BN_MILLION,
+        "StableSwap",
+      ),
+    ),
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(token3, BN_MILLION, token4, BN_MILLION, "StableSwap"),
+    ),
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(token4, BN_MILLION, token5, BN_MILLION, "StableSwap"),
+    ),
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(token5, BN_MILLION, token6, BN_MILLION),
+    ),
+  );
+  const eventResponse = getEventResultFromMangataTx(poolEvents);
+  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  liqIds = await getStablePoolId(poolEvents);
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      api.tx.foundationMembers.changeKey(oldFounder2),
+    ),
+  ).then((events) => {
+    const res = getEventResultFromMangataTx(events);
+    expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+});
+
+test("User can buy GASP in multiswap operation", async () => {
+  const userBalanceBeforeSwap = await api.query.tokens.accounts(
+    testUser,
+    token1,
+  );
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.multiswapAssetSell(
+        [liqIds[0], liqIds[1]],
+        token1,
+        BN_TEN_THOUSAND,
+        GASP_ASSET_ID,
+        BN_ZERO,
+      ),
+    ),
+  ).then(async (result) => {
+    await waitSudoOperationSuccess(result, "SudoAsDone");
+  });
+  const userBalanceAfterSwap = await api.query.tokens.accounts(
+    testUser,
+    token1,
+  );
+  expect(userBalanceAfterSwap.free).bnEqual(
+    userBalanceBeforeSwap.free.sub(BN_TEN_THOUSAND),
+  );
```

Comment:
reuse a helper class, or the user with wallet refreshing

Reply:
Done